### PR TITLE
Add new DeepBGC package.

### PR DESCRIPTION
**Pull request by prihoda**
Add new DeepBGC package for detecting and classifying Biosynthetic Gene Clusters.

Package is available on PyPi. Using bioconda it will be installed together with HMMER and Prodigal dependencies so that they don't have to be installed manually.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by prihoda** [comment]
@bioconda/core Please review this pull request, let me know if you have any questions 😉

File: recipes/deepbgc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+{% set hash_type = "sha256" %}
+{% set hash_value = "57946da7fe55b0a82b1d9b8737769db62c4fad4bc6ca75766442d36f63086d2d" %}
+
+package:
+  name: '{{ name|lower }}'
+  version: '{{ version }}'
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ file_ext }}
+  '{{ hash_type }}': '{{ hash_value }}'
```
Comment (dpryan79) [inline recipes/deepbgc/meta.yaml]: This is OK, but it doesn't save any space or make it easier to diff. It's usually easier to just put `sha256 = "57946da7fe55b0a82b1d9b8737769db62c4fad4bc6ca75766442d36f63086d2d"` and be done with it.

File: recipes/deepbgc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+build:
+  number: 0
+  entry_points:
+    - deepbgc = deepbgc.main:main
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
+    - python>=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*
```
Comment (dpryan79) [inline recipes/deepbgc/meta.yaml]: Please change this to `>=3` and change `build:` to `host`. You can then add `noarch: python`.

File: recipes/deepbgc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    - setuptools
+    - argparse
+    - biopython>=1.70 # support for structured comments from version 1.70
+    - scikit-learn==0.18.2 # needed for antiSMASH compatibility
```
Comment (dpryan79) [inline recipes/deepbgc/meta.yaml]: You don't need to pin to a particular `scikit-learn`. If you want to install with antismash and it requires a particular version then `conda create -n foo antismash deepbgc` will install an appropriate version. By not pinning it you keep compatibility with any future changes in antismash's dependencies (as well as dependencies of other tools).

File: recipes/deepbgc/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  number: 0
+  entry_points:
+    - deepbgc = deepbgc.main:main
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
+    - python>=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*
+    - setuptools
+    - argparse
```
Comment (dpryan79) [inline recipes/deepbgc/meta.yaml]: Remove `argparse`, it's been built-in since python 2.6

File: recipes/deepbgc/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    - numpy>=1.16.1
+    - keras>=2.2.4
+    - tensorflow==1.12.0
+    - matplotlib>=2.2.3
+    - appdirs>=1.4.3
+    - pytest>=4.3.0
+    - pytest-mock>=1.10.1
+    - appdirs>=1.4.3
+  run:
+    - python>=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*
```
Comment (dpryan79) [inline recipes/deepbgc/meta.yaml]: As above for the python pinning.

File: recipes/deepbgc/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    - deepbgc info
+
+about:
+  home: https://github.com/Merck/DeepBGC
+  license: MIT
+  license_family: MIT
+  summary: DeepBGC - Biosynthetic Gene Cluster detection and classification
+
+extra:
+  recipe-maintainers: ''
```
Comment (dpryan79) [inline recipes/deepbgc/meta.yaml]: Please remove the `extra:` section.

**Comment by dpryan79** [comment]
Thanks @prihoda! Please see my comments.

**Comment by dpryan79** [comment]
Looks good!

**Pull request merged**